common/logisticspipes/gui/guidebook/GuiGuideBook.java

```diff
@@ -0,0 +1,779 @@
+		this.prevPage = this.addButton(new GuiGuideBookTexturedButton(2, area$xCenter - area$offsetCenterX, gui$y3 - 12, gui$arrowWidth, gui$arrowHeight, 0, 0, z$titleButtons, 144, 0, 24, 16, true, GuiGuideBookTexturedButton.EnumButtonType.NORMAL));
+		this.nextPage = this.addButton(new GuiGuideBookTexturedButton(3, area$xCenter + area$offsetCenterX - gui$arrowWidth, gui$y3 - 12, gui$arrowWidth, gui$arrowHeight, 0, 0, z$titleButtons, 168, 0, 24, 16, true, GuiGuideBookTexturedButton.EnumButtonType.NORMAL));
+		for (int i = 0; i < maxTabs; i++) {
+			tabList.add(new GuiGuideBookTabButton(i, gui$x3 - 2 - 2 * gui$tagWidth - (i * gui$tagWidth), gui$y0, gui$tagWidth, gui$tagHeight, ""));
+		}
+		this.updateButtonVisibility();
+	}
+
+	@Override
+	public boolean doesGuiPauseGame() {
+		return false;
+	}
+
+	@Override
+	public void onGuiClosed() {
+		currentPage.setProgress(slider.getProgress());
+		ItemGuideBook.setCurrentPage(currentPage, hand);
+		super.onGuiClosed();
+	}
+
+	@Override
+	protected void actionPerformed(GuiButton button) throws IOException {
+		switch (button.id) {
+			case 1:
+				state = menu;
+				slider.reset();
+				break;
+			case 2:
+				prevPage();
+				break;
+			case 3:
+				nextPage();
+				break;
+			default:
+				break;
+		}
+		updateTitle();
+		updateButtonVisibility();
+	}
+
+	@Override
+	protected void mouseClicked(int mouseX, int mouseY, int mouseButton) throws IOException {
+		if (state == menu)
+			for (MenuItemsDivision div : divisionsList) {
+				for (MenuItem item : div.getList()) {
+					if (item.mousePressed()) {
+						item.playPressSound(this.mc.getSoundHandler());
+						this.pressedItem(item);
+						updateButtonVisibility();
+						return;
+					}
+				}
+			}
+		for (GuiGuideBookTabButton tab : tabList) {
+			if (tab.mousePressed(mc, mouseX, mouseY)) {
+				tab.playPressSound(this.mc.getSoundHandler());
+				if (!tab.isSet && state != menu) {
+					PageInformation setPage = new PageInformation();
+					setPage.page = currentPage.page;
+					setPage.progress = currentPage.progress;
+					if (savedTabs.addTab(new SavedTab(setPage, state))) {
+						tab.isSet = true;
+						if (tab.isSet) tab.isActive = true;
+					}else{
+						return;
+					}
+				} else if(tab.isSet && !tab.isActive){
+					SavedTab info = new SavedTab(savedTabs.tabs.get(tab.id).page, savedTabs.tabs.get(tab.id).state);
+					state = page;
+					currentPage.set(info);
+				}
+				updateButtonVisibility();
+				return;
+			}
+		}
+		super.mouseClicked(mouseX, mouseY, mouseButton);
+		updateButtonVisibility();
+	}
+
+	private void pressedItem(MenuItem item) {
+		selectChapter(item);
+	}
+
+	protected void selectChapter(MenuItem item) {
+		currentPage.set(item);
+		state = page;
+		updateTitle();
+		updateButtonVisibility();
+	}
+
+	protected void updateTitle() {
+		String title = "";
+		title += gbc.getTitle();
+		if (state == menu) title += ": Menu";
```

Comment:
if you have an else with brackets, use brackets on the if, please

Reply:
Surely :)